test/e2e/operator_test.go

```diff
@@ -0,0 +1,436 @@
+		if err != nil {
+			klog.Errorf("Unable to list pods: %v", err)
+			return false, nil
+		}
+		for _, pod := range podItems.Items {
+			// skip if pod.Name _doesn't_ have operatorclient.OperandName (operand should have this)
+			// or if it _has_ operatorclient.OperandName + '-operator'
+			if !strings.HasPrefix(pod.Name, "openshift-cli-manager-") || strings.HasPrefix(pod.Name, "openshift-cli-manager-operator") {
+				continue
+			}
+			klog.Infof("Checking pod: %v, phase: %v, deletionTS: %v\n", pod.Name, pod.Status.Phase, pod.GetDeletionTimestamp())
+			if pod.Status.Phase == corev1.PodRunning && pod.GetDeletionTimestamp() == nil {
+				cliManagerPod = pod.DeepCopy()
+				return true, nil
+			}
+		}
+		return false, nil
+	}); err != nil {
+		klog.Errorf("Unable to wait for the CLI Manager (operand) pod to run")
+		os.Exit(1)
+	}
+
+	klog.Infof("CLI Manager (operand) pod running in %v", cliManagerPod.Name)
+
+	r, err := routeClient.Routes("openshift-cli-manager-operator").Get(context.TODO(), "openshift-cli-manager", metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Unable to get route host: %v", err)
+		os.Exit(1)
+	}
+
+	krewUrl := fmt.Sprintf("https://%s/cli-manager", r.Spec.Host)
+
+	if err := wait.PollUntilContextTimeout(context.TODO(), 5*time.Second, 5*time.Minute, false, func(ctx context.Context) (bool, error) {
+		klog.Infof("checking the route is alive")
+
+		tr := &http.Transport{
+			// Just a simple get request to check the route is up and running.
+			// So that we can use skip tls verification.
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+		client := &http.Client{Transport: tr}
+		resp, err := client.Get(krewUrl)
+		if err != nil {
+			klog.Errorf("Unable to send request to %s: %v", krewUrl, err)
+			return false, nil
+		}
+		defer resp.Body.Close()
+		// we are checking notfound because basically our custom git server
+		// does not serve anything and it is legit to get this error in GET request.
+		// Whereas, by getting NotFound error proves that route configuration is up.
+		if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusNotFound {
+			return true, nil
+		}
+		klog.Infof("still not alive, status code %d", resp.StatusCode)
+		return false, nil
+	}); err != nil {
+		klog.Errorf("Unable to wait for CLI Manager route")
+		os.Exit(1)
+	}
+
+	os.Exit(m.Run())
+}
+
+func TestCLIManager(t *testing.T) {
+	customKrewIndexName := "test-e2e"
+	routeClient := getRouteClient()
+
+	r, err := routeClient.Routes("openshift-cli-manager-operator").Get(context.TODO(), "openshift-cli-manager", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("route get error %v", err)
+	}
+	krewUrl := fmt.Sprintf("https://%s/cli-manager", r.Spec.Host)
+
+	currentPath := homedir.HomeDir() + "/.krew"
+	cmd := exec.Command("oc", "krew", "index", "add", customKrewIndexName, krewUrl)
+	cmd.Env = []string{
+		"GIT_SSL_NO_VERIFY=true",
```

Comment:
Krew executes git commands https://github.com/kubernetes-sigs/krew/blob/6e7a20d181394ce108e408b69e67f57680bcf203/internal/gitutil/git.go#L80 without any configuration for CA bundles. In our case, we have self-signed CA bundles. We have to disable it temporarily in our e2e tests.